### PR TITLE
feat: send donation cancellation email

### PR DESCRIPTION
**Pull request by linnall**
## Implementation description. How did you make this change?
Admin email (recurring donation)
Screen Shot 2022-02-12 at 4 00 42 PM
Donor email (recurring donation)
Screen Shot 2022-02-12 at 4 00 27 PM

Note: please refer to Notion ticket for copy for when **admin deletes donations**

* added `sendEmailVerificationAfterSchedulingADonation` to scheduling service
* call `sendEmailVerificationAfterSchedulingADonation` in `deleteSchedulingById` and `deleteSchedulingByRecurringDonationId`

## Steps to test
1. After checking out branch, in the implementation of `schedulingService.ts`, **replace `TEMP_ADMIN_EMAIL` with YOUR personal email** so we don't spam the actual community fridge account :)
2. Create a one-time donation
2. Cancel it
4. Ensure email is received to donor email (refer to copy on linked Notion ticket)
5. Do steps 2-4 with a recurring donation
6. Do steps 2-5 where admin (account using your uwblueprint email) deletes the schedules

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] The appropriate tests if necessary have been written

### CODE REVIEW

File: backend/typescript/services/interfaces/schedulingService.ts [hunk 1]
```diff
@@ -37,17 +37,33 @@ interface ISchedulingService {
...
+   * @param schedule object that contains information on cancelled donation
+   * @param isRecurringDonation 0 if the cancelled donation is a one-time donation, 1 otherwise
+   * @param isAdminDeleted 0 if donor that scheduled donation is cancelling donation, 1 otherwise
+   * @throws Error if unable to send email
+   */
+  sendEmailAfterSchedulingCancellation(
+    donor: UserDonorDTO,
+    schedule: SchedulingDTO,
+    isRecurringDonation: number,
+    isAdminDeleted: number,
```
Comment (hanlinc27) [inline backend/typescript/services/interfaces/schedulingService.ts:64]: any reason why you chose `number` instead of `boolean`?

Comment (linnall) [reply backend/typescript/services/interfaces/schedulingService.ts:64]: I'm writing in C for class right now XD will switch it for better readability :)

**Review by hanlinc27 — COMMENTED** [review]
Adding changes on the frontend and routes to know whether it's a donor deletion or admin deletion.

**Review by jessiiepeng — APPROVED** [review]
It works for me!

**Pull request merged**